lib/src/pubspec.dart

```diff
@@ -445,6 +446,17 @@ class Pubspec {
   bool get isEmpty =>
       name == null && version == Version.none && dependencies.isEmpty;
 
+  /// The language version implied by the sdk constraint.
+  ///
+  /// Given no or unbounded constraint we assume language version 1.0.
```

Comment:
Is this correct?

Doesn't the spec say that the fallback language version is the whatever the current SDK has?

I can see that this might cause validation errors if you have no SDK constraint..

Reply:
pub.dev treats packages with no constraint as dart 1 packages, so at least for null-safety checking I think this makes sense. We might have to revise later.